corehq/apps/cleanup/migrations/0012_add_es_index_to_checkpoint_ids.py

```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.db import models, migrations
+from corehq.sql_db.operations import HqRunPython
+
+
+RENAMES = (
+    ("applications-to-elasticsearch", "ApplicationToElasticsearchPillow-hqapps_2016-07-11_2128"),
+    ("all-cases-to-elasticsearch", "CaseToElasticsearchPillow-hqcases_2016-03-04"),
+    ("all-xforms-to-elasticsearch", "XFormToElasticsearchPillow-xforms_2016-07-07"),
+    ("case-search-to-elasticsearch", "CaseSearchToElasticsearchPillow-case_search_2016-03-15"),
+    ("GroupPillow", "GroupPillow-hqgroups_20150403_1501"),
+    ("GroupToUserPillow", "GroupToUserPillow-hqusers_2016-07-19"),
+    ("KafkaDomainPillow", "KafkaDomainPillow-hqdomains_2016-08-08"),
+    ("ledger-to-elasticsearch", "LedgerToElasticsearchPillow-ledgers_2016-03-15"),
+    ("report-cases-to-elasticsearch", "ReportCaseToElasticsearchPillow-report_cases_czei39du507m9mmpqk3y01x72a3ux4p0"),
+    ("report-xforms-to-elasticsearch", "ReportXFormToElasticsearchPillow-report_xforms_20160707_2322"),
+    ("sql-sms-to-es", "SqlSMSPillow-smslogs_708c77f8e5fe00286fa5791e9fa7d45f"),
+    ("UnknownUsersPillow", "UnknownUsersPillow-hqusers_2016-07-19"),
+    ("UserPillow", "UserPillow-hqusers_2016-07-19"),
+)
+
+
+def copy_checkpoints(apps, schema_editor):
+    from pillowtop import get_all_pillow_instances
+    DjangoPillowCheckpoint = apps.get_model('pillowtop', 'DjangoPillowCheckpoint')
+    current_checkoint_ids = set(
+        [pillow_instance.checkpoint.checkpoint_id for pillow_instance in get_all_pillow_instances()]
+    )
+
+    for old_id, new_id in RENAMES:
+        try:
+            checkpoint = DjangoPillowCheckpoint.objects.get(checkpoint_id=old_id)
+            # since checkpoint_id is the primary key, this should make a new model
+            # which is good in case we need to rollback
+            checkpoint.checkpoint_id = new_id
+            checkpoint.save()
+            assert new_id in current_checkoint_ids
+        except DjangoPillowCheckpoint.DoesNotExist:
+            print 'warning: legacy pillow checkpoint with ID {} not found'.format(old_id)
+        except AssertionError:
+            print 'warning: no current pillow found with checkpoint ID {}'.format(new_id)
+
+
+def delete_checkpoints(apps, schema_editor):
+    DjangoPillowCheckpoint = apps.get_model('pillowtop', 'DjangoPillowCheckpoint')
+    for _, new_id in RENAMES:
```

Comment:
I think this was merged without being updated. Doubt that the reverse migration is a big deal though